base/compiler/ssair/verify.jl

```diff
@@ -96,31 +98,49 @@ end
 
 function verify_ir(ir::IRCode, print::Bool=true,
                    allow_frontend_forms::Bool=false,
-                   𝕃ₒ::AbstractLattice = SimpleInferenceLattice.instance)
+                   𝕃ₒ::AbstractLattice = SimpleInferenceLattice.instance,
+                   mi::Union{Nothing,MethodInstance}=nothing)
+    function raise_error()
+        error_args = Any["IR verification failed."]
+        if isdefined(Core, :Main) && isdefined(Core.Main, :Base)
+            # ensure we use I/O that does not yield, as this gets called during compilation
+            firstline = invokelatest(Core.Main.Base.IRShow.debuginfo_firstline, ir.debuginfo)
+        else
+            firstline = nothing
+        end
+        if firstline !== nothing
+            file, line = firstline
+            push!(error_args, "\n", "  Code location:   ", file, ":", line)
+        end
+        if mi !== nothing
+            push!(error_args, "\n", "  Method instance: ", mi)
+        end
+        error(error_args...)
```

Comment:
Maybe we could make it a "CompilationError" and capture the method_instance and then use the show error infrastructure for the pretty rendering.

Reply:
Hm, even during bootstrapping `error` calls `throw(ErrorException(...))`, where is the printing of that special-cased?

Reply:
It does call `throw(ErrorException(...))` although the it will be printed via `jl_static_show` since `show` isn't available.